moveit_core/robot_model/src/link_model.cpp

```diff
@@ -85,25 +85,25 @@ void moveit::core::LinkModel::setGeometry(const std::vector<shapes::ShapeConstPt
          collision_origin_transform_[i].translation().norm() < std::numeric_limits<double>::epsilon()) ?
             1 :
             0;
-    Eigen::Vector3d ei = shapes::computeShapeExtents(shapes_[i].get());
-    Eigen::Vector3d p1 = collision_origin_transform_[i] * (-ei / 2.0);
-    Eigen::Vector3d p2 = collision_origin_transform_[i] * (-p1);
+    Eigen::Vector3d extents = shapes::computeShapeExtents(shapes_[i].get());
+    Eigen::Affine3d transform = collision_origin_transform_[i];
 
-    if (i == 0)
+    if (shapes_[i]->type != shapes::MESH)
     {
-      a = p1;
-      b = p2;
+      aabb.extendWithTransformedBox(transform, extents);
     }
     else
     {
-      for (int i = 0; i < 3; ++i)
-        a[i] = std::min(a[i], p1[i]);
-      for (int i = 0; i < 3; ++i)
-        b[i] = std::max(b[i], p2[i]);
+      const shapes::Mesh* mesh = dynamic_cast<const shapes::Mesh*>(shapes_[i].get());
```

Comment:
Can you add a comment, why meshes are handled differently?
I guess, with your approach, you compute the smallest AABB for the mesh given the transformation, while otherwise an AABB of the mesh (in default pose) is computed and transformed afterwards, which usually gives a larger box.

Reply:
You're right it deserves a comment, since the reasoning is different -  `shapes::computeShapeExtents` only gives the extents and doesn't provide info about the position of the center of the mesh in these extents.